index.js

```diff
@@ -186,6 +186,14 @@ function rateLimitRequestHandler (params, pluginComponent) {
     let current = 0
     let ttl = 0
 
+    let maximum
+
+    if (typeof params.max === 'number' && !isNaN(params.max)) {
+      maximum = params.max
+    } else {
+      maximum = await params.max(req, key)
+    }
+
```

Comment:
Would this break if moving the check before rather than after? it shouldn't... we don't make any promises when the function would be executed...